### PR TITLE
Add appendPageQueryParam parameter

### PR DESCRIPTION
**Pull request by benbristow**
Adds an option on each SitemapNode to disable the concatenation of the `?page=` param to each `loc`

### CODE REVIEW

File: tests/SimpleSiteMap.Tests/ConvertToSiteMapTests.cs [hunk 1]
```diff
@@ -9,33 +9,48 @@ namespace SimpleSiteMap.Tests
...
+        private static readonly SitemapService SitemapService = new SitemapService();
+        
         [Fact]
         public void GivenDataWith10Items_ConvertToSiteMap_CreatesASitemapResult()
         {
             // Arrange.
-            var startDate = DateTime.SpecifyKind(new DateTime(2014, 11, 21, 18, 58, 00), DateTimeKind.Utc);
-            var data = SimpleSiteMapHelpers.CreateFakeSitemapNodes(100, startDate).ToPartition(10);
-            var siteMapService = new SitemapService();
+            var data = SimpleSiteMapHelpers.CreateFakeSitemapNodes(100, StartDate).ToPartition(10);
```
Comment (PureKrome) [inline tests/SimpleSiteMap.Tests/ConvertToSiteMapTests.cs:20]: Ah! that reminds me - i need to remove this extension and leverage MoreLinq, for this in the tests. Ta!!!

File: .gitignore [hunk 2]
```diff
@@ -38,4 +38,5 @@ App_Data/
 *.ncrunchsolution
 *.nupkg
 .vs
-*.orig
\ No newline at end of file
+*.orig
```
[the remark this answers is not in the document]
Comment (benbristow) [reply .gitignore:41]: No idea, but I think that was there before.

**Comment by benbristow** [comment]
> Wow nice stuff! _really_ appreciate the PR @benbristow ! 👏
> 
> > Adds an option on each SitemapNode to disable the concatenation of the ?page= param to each loc
> 
> Now I understand the code .. but I'm missing one piece of info: what's the scenario where you don't want `?page=<...>` ? Is that if you have only _one_ page?

Mainly if you've got a ?page= query param already there for pagination (or other query params there as it'd look a bit weird like `mypage/?foo=bar?page=2`

I've never seen a sitemap with the ?page= query param before added to the `loc` entries.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (PureKrome) [reply .gitignore:41]: ha oh yeah. it's a weird 'changed' line.

[the remark this answers is not in the document]
Comment (PureKrome) [reply .gitignore:41]: oh wait. i ment -> what is an `.idea/` folder? that's something u added.

File: src/SimpleSitemap/SitemapService.cs [hunk 3]
```diff
@@ -81,6 +82,18 @@ private static XElement CreateXmlSitemapIndex(ICollection<SitemapNode> sitemapNo
             }
 
             return root;
+            
+            XElement GenerateLocXElement(SitemapNode sitemapNode, int page, bool appendPageQueryParam)
+            {
+                var uriString = sitemapNode.Url.ToString();
```
[the remark this answers is not in the document]
Comment (PureKrome) [reply src/SimpleSitemap/SitemapService.cs:88]: thoughts on this @benbristow ?

[the remark this answers is not in the document]
Comment (benbristow) [reply src/SimpleSitemap/SitemapService.cs:88]: Apologies. Never got time to look into this. Does the XML library not automatically escape the URL anyway?

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (PureKrome) [reply .gitignore:41]: AH! is this related to JETBRAINS IntelliJ ?

File: src/SimpleSitemap/SitemapService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (PureKrome) [reply src/SimpleSitemap/SitemapService.cs:88]: TIL. it's the `Uri` class that handles this, automatically, it seems.

```
var url = new Uri("https://www.google.com.au/some/url/foo?name=fred jane bob&age=19&json=[{\"fred\"=\"true\"}]");
```

And then adding this to the `node` (which is just a class with a `Uri` property) ... and then checking the value of the `Uri` property ....

image

which then means ...

```
<?xml version="1.0" encoding="utf-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  <url>
    <loc>https://www.google.com.au/some/url/foo?name=fred%2520jane%2520bob&amp;age=19&amp;json=[%257B%2522fred%2522=%2522true%2522%257D]</loc>
    <lastmod>2020-05-02T11:57:00+00:00</lastmod>
    <changefreq>daily</changefreq>
    <priority>0.5</priority>
  </url>
...
```

so yep! it's auto handled/encoded. super sweet.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (PureKrome) [reply .gitignore:41]: yes please @benbristow - then we can merge this PR. 👍

[the remark this answers is not in the document]
Comment (PureKrome) [reply .gitignore:41]: :wave: very polite ping :)

[the remark this answers is not in the document]
Comment (benbristow) [reply .gitignore]: Sorry, never received this ping. Updated.

**Comment by benbristow** [comment]
I see the tests have failed. I can't seem to reproduce this locally after re-running in Visual Studio many times. I think this is a blip. Are you able to re-run the tests on Appveyor?

**Comment by PureKrome** [comment]
Error seems related to a missing file. I'll try and pull down the branch and test that locally, later.

watch this space!

**Comment by PureKrome** [comment]
Done @benbristow :) I fixed the issue (ref 83e12a2) and merged. 

NuGet has also been updated. It's now version 3.1.0.

Thanks heaps for the PR - really appreciate it.

**Pull request merged**